server/publications/collections/product-publications.app-test.js

```diff
@@ -14,18 +14,37 @@ import { RevisionApi } from "/imports/plugins/core/revisions/lib/api/revisions";
 Fixtures();
 
 describe("Publication", function () {
-  const shopId = Random.id();
-  const primaryShopId = Random.id();
+  let shopId;
+  let merchantShopId;
+  let primaryShopId;
   let sandbox;
 
+  let merchantShop1ProductIds;
+  let merchantShop2ProductIds;
+  let primaryShopProductIds;
+  let allProductIds;
+  let activeMerchantProductIds;
+
+  const productScrollLimit = 24;
+
   beforeEach(function () {
+    shopId = Random.id();
+    merchantShopId = Random.id();
+    primaryShopId = Random.id();
```

Comment:
I was occasionally getting a duplicate shop id error, so I moved these to a before block.